alpakaConfig.cmake

```diff
@@ -134,7 +134,7 @@ IF(${ALPAKA_DEBUG} GREATER 1)
     SET(Boost_DETAILED_FAILURE_MSG ON)
 ENDIF()
 IF(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE)
-    FIND_PACKAGE(Boost ${_ALPAKA_BOOST_MIN_VER} QUIET COMPONENTS fiber context system thread chrono date_time)
+    FIND_PACKAGE(Boost ${_ALPAKA_BOOST_MIN_VER} QUIET COMPONENTS fiber context system thread atomic chrono date_time)
```

Comment:
Are you really sure that atomic is necessary? See https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#814

Reply:
yes, transitive [via `Boost.Thread`](https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L823)

Reply:
Then this is wrong in CMake see here: [1.53 doc](https://www.boost.org/doc/libs/1_53_0/doc/html/thread/build.html) or [1.66 doc](https://www.boost.org/doc/libs/1_66_0/doc/html/thread/build.html) and one of the last builds
```
Apr 06 09:09:36 Component configuration:
Apr 06 09:09:36 
Apr 06 09:09:36     - atomic                   : not building
Apr 06 09:09:36     - chrono                   : building
Apr 06 09:09:36     - container                : not building
Apr 06 09:09:36     - context                  : building
Apr 06 09:09:36     - coroutine                : not building
Apr 06 09:09:36     - coroutine2               : not building
Apr 06 09:09:36     - date_time                : building
Apr 06 09:09:36     - exception                : not building
Apr 06 09:09:36     - fiber                    : building
Apr 06 09:09:36     - filesystem               : not building
Apr 06 09:09:36     - graph                    : not building
Apr 06 09:09:36     - graph_parallel           : not building
Apr 06 09:09:36     - iostreams                : not building
Apr 06 09:09:36     - locale                   : not building
Apr 06 09:09:36     - log                      : not building
Apr 06 09:09:36     - math                     : not building
Apr 06 09:09:36     - metaparse                : not building
Apr 06 09:09:36     - mpi                      : not building
Apr 06 09:09:36     - program_options          : not building
Apr 06 09:09:36     - python                   : not building
Apr 06 09:09:36     - random                   : not building
Apr 06 09:09:36     - regex                    : not building
Apr 06 09:09:36     - serialization            : not building
Apr 06 09:09:36     - signals                  : not building
Apr 06 09:09:36     - system                   : building
Apr 06 09:09:36     - test                     : building
Apr 06 09:09:36     - thread                   : building
Apr 06 09:09:36     - timer                    : not building
Apr 06 09:09:36     - type_erasure             : not building
Apr 06 09:09:36     - wave                     : not building
```

Reply:
Interesting, but the linked 1.66 docs above state:

>  Boost.Thread uses by default Boost.Atomic in POSIX platforms to implement call_once..

Reply:
No, this is a linux build. So at least the boost build system treats it as an optional dependency.

Reply:
Currently, `Boost::atomic` is not built. The possibly incorrect transitive dependency of the `Boost::thread` target seems to require `Boost::atomic` on some builds but it can not be found and the build fails.
So we may need to build `Boost::atomic` when using the CMake targets.

Reply:
Yes, `Boost::thread` in CMake seems to just conservatively assume you might want to use all of it's functionality - which includes the `call_once` functionality we are not using. Anyway, we can just ad it. It does not add much to what it already pulls with `Boost.fiber`.